pyanaconda/network.py

```diff
@@ -272,6 +272,55 @@ def unset(self, *args):
         ifcfglog.debug("IfcfgFile.unset %s: %s", self.filename, args)
         SimpleConfigFile.unset(self, *args)
 
+def ensure_single_initramfs_connections():
+    """Ensure device configured in initramfs has no more than one NM connection.
+
+    In case of multiple connections for device having ifcfg configuration from
+    boot options, the connection should correspond to the ifcfg file.
+    NetworkManager can be generating additional in-memory connection in case it
+    fails to match device configuration to the ifcfg (#1433891).  By
+    reactivating the device with ifcfg connection the generated in-memory
+    connection will be deleted by NM.
+
+    Don't enforce on slave devices for which having multiple connections can be
+    valid (slave connection, regular device connection).
+    """
+
+    rv = []
+
+    for dev_name in nm.nm_devices():
+        try:
+            nm.nm_device_setting_value(dev_name, "connection", "uuid")
+        except nm.SettingsNotFoundError:
+            pass
+        except nm.MultipleSettingsFoundError:
+            if nm.nm_device_is_slave(dev_name):
+                continue
+            ifcfg_path = find_ifcfg_file_of_device(dev_name)
+            if ifcfg_path:
```

Comment:
Hm, I'd like to keep the outer function for networkInitialize readability, but I can't come with any reasonable name for the inner function (which tells something about the nature of the special-purpose tweak we are doing here). Also maybe for better place what to include in the function.

ensure_device_has_connection_from_ifcfg_if_the_ifcfg_was_generated_from_bootoptions

The point about testability seems valid to me in general.
I'll have to think about it. And perhaps, with handling multiple devices better we don't even need the whole function at all. Especially given that it would reactivate the device unconditionally with the risk of invalidating possible existing nfs mount via the connection.